tests/queries/0_stateless/02421_explain_subquery.sql

```diff
@@ -21,3 +23,27 @@ SELECT rows > 1000 FROM (EXPLAIN ESTIMATE SELECT sum(a) FROM t1);
 SELECT count() == 1 FROM (EXPLAIN ESTIMATE SELECT sum(a) FROM t1);
 
 DROP TABLE IF EXISTS t1;
+
+SET allow_experimental_analyzer = 1;
+
+SELECT count() > 3 FROM (EXPLAIN PIPELINE header = 1 SELECT * FROM system.numbers ORDER BY number DESC) WHERE explain LIKE '%Header: system.numbers.number__ UInt64%';
+SELECT count() > 0 FROM (EXPLAIN PLAN SELECT * FROM system.numbers ORDER BY number DESC) WHERE explain ILIKE '%Sort%';
+SELECT count() > 0 FROM (EXPLAIN SELECT * FROM system.numbers ORDER BY number DESC) WHERE explain ILIKE '%Sort%';
+SELECT count() > 0 FROM (EXPLAIN CURRENT TRANSACTION);
+SELECT count() == 1 FROM (EXPLAIN SYNTAX SELECT number FROM system.numbers ORDER BY number DESC) WHERE explain ILIKE 'SELECT%';
+
+-- Note: here we have `Identifier number` instead of `Asterisk` because it's resolved in analyzer.
+-- Without subsquery it still would be `Asterisk`, because in that case analyzer it's not used by InterpreterExplainQuery.
```

Comment:
Not sure if I understand this comment. Consider to add an example.

Reply:
I've changed a comment to outline that it's connected with analyzing query arguments of table functions